engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/callcaching/CallCacheDiffQueryParameter.scala

```diff
@@ -0,0 +1,52 @@
+package cromwell.engine.workflow.lifecycle.execution.callcaching
+
+import cats.data.{NonEmptyList, Validated}
+import cats.implicits._
+import cromwell.engine.workflow.lifecycle.execution.callcaching.CallCacheDiffQueryParameter.CallCacheDiffQueryCall
+
+import scala.util.{Failure, Success, Try}
+
+object CallCacheDiffQueryParameter {
+  case class CallCacheDiffQueryCall(workflowId: String, callFqn: String, jobIndex: Option[Int])
+
+  private def missingWorkflowError(attribute: String) = s"missing $attribute query parameter".invalidNel
+
+  private def extractAttribute(parameters: Seq[(String, String)], parameter: String): Validated[NonEmptyList[String], String] = {
```

Comment:
ToL: This and `extractIndex` could be declared locally within `fromParameters`, saving you from declaring this arg and passing in the same params everywhere